cli/src/main/java/org/eclipse/hono/cli/client/ApplicationClient.java

```diff
@@ -0,0 +1,351 @@
+                    return;
+                }
+                //create the connection
+                try{
+                    clientFactory = ApplicationClientFactory.create(HonoConnection.newConnection(vertx, fork.honoClientConfig));
+                    latch = new CountDownLatch(1);
+                    final Receiver receiverInstance = new Receiver(clientFactory, vertx, fork);
+                    receiverInstance.start(latch);
+                    //to handle forced interruption
+                    Signal.handle(new Signal("INT"), new SignalHandler() {
+                        // Signal handler method
+                        public void handle(final Signal signal) {
+                            latch.countDown();
+                        }
+                    });
+                    // if everything went according to plan, the next step will block forever
+                    latch.await();
+                } catch(final Exception e) {
+                    log.error("Error occurred during initialization of connection: {}", e.getMessage());
+                }
+            } catch (final CloneNotSupportedException e) {
+                System.out.println("Unable to clone the configuration: "+e.getMessage());
+            }
+        }
+
+        @ShellMethod("Send a command\n" +
+                "\t\t\t\t[-H --host] [-P --port] [-u --username] [-p --password] [-T --tenant]")
+        void sendCommand(@ShellOption(value={"-H", "--host"}, defaultValue="", help="The address that the AMQP_NETWORK is running on") final String host,
+                     @ShellOption(value={"-P", "--port"}, defaultValue="0", help="The port that the AMQP_NETWORK is listening") final int port,
+                     @ShellOption(value={"-u", "--username"}, defaultValue="", help="Credentials to use to connect to the upperbound endpoint") final String username,
+                     @ShellOption(value={"-p", "--password"}, defaultValue="", help="Credentials to use to connect to the upperbound endpoint") final String password,
+                     @ShellOption(value={"-T", "--tenant"}, defaultValue="", help="The tenant you want to connect to") final String tenant){
+            try {
+                final ClientConfig fork = (ClientConfig) clientConfig.clone();
+                if (!setupConfiguration(fork, host, port, username, password, tenant, "")) {
+                    return;
+                }
+                //create the connection
+                try {
+                    clientFactory = ApplicationClientFactory.create(HonoConnection.newConnection(vertx, fork.honoClientConfig));
+                    latch = new CountDownLatch(1);
+                    final Commander commanderInstance = new Commander(clientFactory, vertx, fork, inputReader);
+                    commanderInstance.start(latch);
+                    //to handle forced interruption
+                    Signal.handle(new Signal("INT"), new SignalHandler() {
```

Comment:
Can you explain what you are trying to achieve here?

Reply:
For insight in the shell, here you can find all: https://projects.spring.io/spring-shell/
However the basic knowledge is that the methods annotated as a `@Command` will be executed as a command in the shell.

> what you are trying to achieve here?

If you run the command "receiver" for example and then want to stop it, pressing CTRL+C, you'll exit the entire shell but the expected behaviour is to end just the execution of the command. (and then if you want you can exit the shell)

Reply:
FMPOV that is not important. Let's go with the default behavior in a first step, i.e. when pressing CTRL + C, the shell terminates. In practice, you will need to run two shells anyway, one for receiving messages and another one for interactive tasks ...